rust/src/treefile.rs

```diff
@@ -2755,7 +2755,12 @@ impl TreeComposeConfig {
     fn hasher_update(&self, hasher: &mut glib::Checksum) -> Result<()> {
         // don't use pretty mode to increase the chances of a stable serialization
         // https://github.com/projectatomic/rpm-ostree/pull/1865
-        hasher.update(serde_json::to_vec(self)?.as_slice());
+        let layers = serde_json::to_string(&self.base.ostree_layers.clone()).unwrap();
```

Comment:
We generally avoid using `.unwrap()` - it will cause the process to hard crash (dump core) if it's hit.  In this case, the serialization shouldn't normally fail, but prefer using `?` for cases like this anyways.